src/js/game/hud/parts/mass_selector.js

```diff
@@ -288,49 +309,61 @@ export class HUDMassSelector extends BaseHUDPart {
-                    if (contents && this.root.logic.canDeleteBuilding(contents)) {
-                        // Prevent rendering the overlay twice
-                        const uid = contents.uid;
-                        if (renderedUids.has(uid)) {
-                            continue;
+                    let entities = [];
+                    if (this.multiLayerSelect) {
+                        entities = this.root.map.getLayersContentsMultipleXY(x, y);
+                    } else {
+                        entities = [this.root.map.getLayerContentXY(x, y, this.root.currentLayer)];
+                    }
+
+                    for (let i = 0; i < entities.length; ++i) {
+                        let entity = entities[i];
+                        if (entity && this.root.logic.canDeleteBuilding(entity)) {
+                            // Prevent rendering the overlay twice
+                            const uid = entity.uid;
+                            if (renderedUids.has(uid)) {
+                                continue;
+                            }
+                            renderedUids.add(uid);
+
+                            this.RenderSelectonPreviewTile(parameters, entity);
                         }
-                        renderedUids.add(uid);
-
-                        const staticComp = contents.components.StaticMapEntity;
-                        const bounds = staticComp.getTileSpaceBounds();
-                        parameters.context.beginRoundedRect(
-                            bounds.x * globalConfig.tileSize + boundsBorder,
-                            bounds.y * globalConfig.tileSize + boundsBorder,
-                            bounds.w * globalConfig.tileSize - 2 * boundsBorder,
-                            bounds.h * globalConfig.tileSize - 2 * boundsBorder,
-                            2
-                        );
-                        parameters.context.fill();
                     }
                 }
             }
         }
 
-        parameters.context.fillStyle = THEME.map.selectionOverlay;
-        this.selectedUids.forEach(uid => {
-            const entity = this.root.entityMgr.findByUid(uid);
-            const staticComp = entity.components.StaticMapEntity;
-            const bounds = staticComp.getTileSpaceBounds();
-            parameters.context.beginRoundedRect(
-                bounds.x * globalConfig.tileSize + boundsBorder,
-                bounds.y * globalConfig.tileSize + boundsBorder,
-                bounds.w * globalConfig.tileSize - 2 * boundsBorder,
-                bounds.h * globalConfig.tileSize - 2 * boundsBorder,
-                2
-            );
-            parameters.context.fill();
-        });
+		//EXTREMELY SLOW. There must be a better way. (Possibly use a Array)
+		for(let i = 0; i < this.selectedEntities.length; ++ i){
+			const entity = this.selectedEntities[i];
+			this.RenderSelectonPreviewTile(parameters, entity);

```

Comment:
I've looked over some of the other components and it seems fairly normal to render things in this way.